apiserver/facades/client/client/filtering.go

```diff
@@ -280,12 +281,45 @@ func buildUnitMatcherShims(u *state.Unit, patterns []string) []closurePredicate
 	}
 }
 
+// portsFromString gets "from port" and "to port" value from port string.
+func getPortsFromString(portStr string) (int, int, error) {
+	var portFrom, portTo int
+	var err error
+	portFromStr, portToStr, isPortRange := strings.Cut(portStr, "-")
+	if isPortRange {
+		portFrom, err = strconv.Atoi(portFromStr)
+		if err != nil {
+			return -1, -1, err
+		}
+		portTo, err = strconv.Atoi(portToStr)
+		if err != nil {
+			return -1, -1, err
+		}
+	} else {
+		portFrom, err = strconv.Atoi(portStr)
+		if err != nil {
+			return -1, -1, err
```

Comment:
Knitpick, but I think returning -1 as an error value is more of a python thing, whereas in Go we generally return 0 (the default/'empty' int value), along with a separate error